### PR TITLE
Remove deprecated "deft-extension" 

### PR DESCRIPTION
**Pull request by mclearc**
- add setting for recognizing multiple extensions
- modify with (setq deft-extensions '("your extension here")) in you config file

### CODE REVIEW

**Comment by robbyoconnor** [comment]
Should prolly rebase.

File: layers/deft/README.org [hunk 1]
```diff
@@ -16,7 +16,12 @@ browsing and creating notes.
 ** Differences from default
 
 This layer sets the default to use filenames for note titles as well as
-=org-mode= for editing. The default extension is still =.txt= though.
+=org-mode= for editing. The default extension is =org=. Configure by adding the
+following in your config file: 
+
+#+Begin_SRC emacs-lisp
+  (setq deft-extensions '("your" "extension" "here"))  
+#+END_SRC
```
Comment (robbyoconnor) [inline layers/deft/README.org]: Change to: 
+#+Begin_SRC emacs-lisp
+  (setq deft-extensions '("org" "md" "txt"))  
+#+END_SRC

**Comment by mclearc** [comment]
Done.

**Comment by robbyoconnor** [comment]
:+1: :100:

**Comment by robbyoconnor** [comment]
Should state that other extensions are possible, but you are offering sane defaults for copy/pasting :)

**Comment by robbyoconnor** [comment]
Awesome, now rebase and squash ;) Other than that nice :D :100:

**Comment by robbyoconnor** [comment]
You mucked up the rebase dude ;)

**Comment by mclearc** [comment]
I did! Sorry, I'm new to all this. Will fix it.

**Comment by robbyoconnor** [comment]
interactive rebase dude and choose `fixup` for non-interesting commits -- basically all but one will be `fixup` :)

**Comment by mclearc** [comment]
Hopefully that's fixed it.

**Comment by robbyoconnor** [comment]
Yup :+1: :100:

**Comment by syl20bnr** [comment]
Thank you :+1: 
Cherry-picked in develop branch.

**Pull request closed without merging**